src/Symfony/Component/Config/MetadataValidatorInterface.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Config;
+
+/**
+ * Interface for MetadataValidators.
+ *
+ * When a ConfigCache instance is checked for freshness, all its associated
+ * metadata resources are passed to MetadataValidators. The MetadataValidators
+ * can then inspect the resources and decide whether the cache can be considered
+ * fresh or not.
+ *
+ * @author Benjamin Klotz <bk@webfactory.de>
+ */
+interface MetadataValidatorInterface
```

Comment:
I'm open for naming suggestions. 

* `CacheChecker`
* `CacheMetadataChecker`
* `MetadataChecker`
* `ResourceChecker`
* `ResourceMetadataChecker`

...?